interp/constrintern.ml

```diff
@@ -2760,16 +2760,16 @@ let interp_type_evars_impls ?(flags=Pretyping.all_no_fail_flags) env sigma ?(imp
 
 (* Not all evars expected to be resolved, with side-effect on evars *)
 
-let interp_constr_evars_gen ?(program_mode=false) env sigma ?(impls=empty_internalization_env) expected_type c =
+let interp_constr_evars_gen ?flags ?(program_mode=false) env sigma ?(impls=empty_internalization_env) expected_type c =
   let c = intern_gen expected_type ~impls env sigma c in
-  let flags = { Pretyping.all_no_fail_flags with program_mode } in
+  let flags = Option.default { Pretyping.all_no_fail_flags with program_mode } flags in
```

Comment:
Not required to be changed in this PR, but it's a semantic footgun to have two optional flags that are not orthogonal.